### PR TITLE
[ENG-6419] Share integration

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-6419]
-   Feature flag: n/a

## Purpose
- Make sure institution dashboard pages talking to SHARE don't break

## Summary of Changes
- Update project page controller's getters to account for missing values
- Add new package `osfmap/jsonld.ts` with utilities to make getting stuff from SHARE easier and more readable

## QA Notes

[ENG-6419]: https://openscience.atlassian.net/browse/ENG-6419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/packages/osfmap/jsonld.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+}
+
+export function getOsfmapValues(osfmapObject: any, propertyPath: string[]) {
+    return Array.from(iterOsfmapValues(osfmapObject, propertyPath));
+}
+
+export function getSingleOsfmapValue(osfmapObject: any, propertyPath: string[]) {
+    return iterOsfmapValues(osfmapObject, propertyPath).next().value;
+}
```
Comment (brianjgeiger) [inline app/packages/osfmap/jsonld.ts:25]: What happens if you use this on something that's not a single value? Does it fail gracefully? Or does typescript help here?

**Comment by futa-ikeda** [comment]
Note: Since this PR is blocking some initial QA efforts, I plan on changing some of the getters in `search-result.ts` to use the new helper functions in a subsequent PR.

**Review by brianjgeiger — APPROVED** [review]
One question about a place that might stand to be more defensive?

File: app/packages/osfmap/jsonld.ts (hunk 1 above)
Comment (futa-ikeda) [reply app/packages/osfmap/jsonld.ts:25]: It should just get the first one, and if there's no `@id` or `@value`, it should just return `undefined`

**Pull request merged**